ext/dom/php_dom.stub.php

```diff
@@ -2,1011 +2,1223 @@
+         */
+        public function prepend(...$nodes): void {}
 
-    /** @tentative-return-type */
-    public function getNamedItem(string $qualifiedName): ?DOMNode {} // TODO DOM spec returns DOMAttr
+        /** @param Node|string $nodes */
+        public function replaceChildren(...$nodes): void {}
+    }
 
-    /** @tentative-return-type */
-    public function getNamedItemNS(?string $namespace, string $localName): ?DOMNode {} // TODO DOM spec returns DOMAttr
+    final class HTMLDocument extends DOM\Document
+    {
+        private function __construct() {}
 
-    /** @tentative-return-type */
-    public function item(int $index): ?DOMNode {} // TODO DOM spec returns DOMAttr
+        public static function createEmpty(string $encoding = "UTF-8"): HTMLDocument {}
 
-    /** @tentative-return-type */
-    public function count(): int {}
+        public static function createFromFile(string $path, int $options = 0, ?string $override_encoding = null): HTMLDocument {}
 
-    public function getIterator(): Iterator {}
-}
+        public static function createFromString(string $source, int $options = 0, ?string $override_encoding = null): HTMLDocument {}
 
-class DOMEntity extends DOMNode
-{
-    /** @readonly */
-    public ?string $publicId;
+        /** @implementation-alias DOMDocument::saveXML */
+        public function saveXML(?\DOMNode $node = null, int $options = 0): string|false {}
 
-    /** @readonly */
-    public ?string $systemId;
+        /** @implementation-alias DOMDocument::save */
+        public function saveXMLFile(string $filename, int $options = 0): int|false {}
 
-    /** @readonly */
-    public ?string $notationName;
+        public function saveHTML(?\DOMNode $node = null): string|false {}
 
-    /**
-     * @readonly
-     * @deprecated
-     */
-    public ?string $actualEncoding = null;
+        public function saveHTMLFile(string $filename): int|false {}
+    }
 
-    /**
-     * @readonly
-     * @deprecated
-     */
-    public ?string $encoding = null;
+    final class XMLDocument extends DOM\Document
+    {
+        /** @implementation-alias DOM\HTMLDocument::__construct */
+        private function __construct() {}
 
-    /**
-     * @readonly
-     * @deprecated
-     */
-    public ?string $version = null;
-}
+        public static function createEmpty(string $version = "1.0", string $encoding = "UTF-8"): XMLDocument {}
 
-class DOMEntityReference extends DOMNode
-{
-    public function __construct(string $name) {}
-}
+        public static function createFromFile(string $path, int $options = 0, ?string $override_encoding = null): XMLDocument {}
 
-class DOMNotation extends DOMNode
-{
-    /** @readonly */
-    public string $publicId;
+        public static function createFromString(string $source, int $options = 0, ?string $override_encoding = null): XMLDocument {}
 
-    /** @readonly */
-    public string $systemId;
-}
+        /** @readonly */
+        public ?string $xmlEncoding;
```

Comment:
Can this be actually `readonly` or don't stubs support this?

Reply:
The difference between `@readonly` and `readonly` is that the former actually means "user can't set this" while the latter means "no one can set this". It is possible for xmlEncoding to change, but not by the user.